packages/core/apps/core-apps-server-internal/src/core_app.ts

```diff
@@ -300,7 +300,10 @@ export class CoreAppsService {
             await soClient.create(DYNAMIC_CONFIG_OVERRIDES_SO_TYPE, newGlobalOverrides, {
               id: DYNAMIC_CONFIG_OVERRIDES_SO_ID,
               overwrite: true,
+              refresh: false,
             });
+            // set it again in memory in case the timer polling the SO for updates has overridden it during this update.
+            this.configService.setDynamicConfigOverrides(req.body);
```

Comment:
The actual change: set it again in memory after updating the SO just in case the poller kicks in while `soClient.create` is running and reverts the configuration this API request intends to set.